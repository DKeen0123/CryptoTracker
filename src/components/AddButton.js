import React from 'react';

const AddButton = props => {
  return <button>{props.label}</button>;
};

export default AddButton;