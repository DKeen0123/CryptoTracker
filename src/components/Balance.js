import React from 'react';

const Balance = () => {
  return (
    <div>
      <h4>Your balance: </h4>
    </div>
  );
};

export default Balance;
