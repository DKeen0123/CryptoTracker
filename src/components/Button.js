import React from 'react';

const Button = () => {
  return <button>Deposit</button>;
};

export default Button;
