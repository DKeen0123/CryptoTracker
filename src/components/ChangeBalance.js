import React from 'react';

const ChangeBalance = () => {
  return (
    <div>
      <input />
    </div>
  );
};

export default ChangeBalance;
