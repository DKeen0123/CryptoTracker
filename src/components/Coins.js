import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/action_bitcoin';
import { fetchEthereum } from '../actions/action_ethereum';

export class Coins extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
    this.props.fetchEthereum();
  }

  calculateBitcoin() {
    const { bitcoin } = this.props;
    if (Object.keys(bitcoin).length === 0) return '';
    return this.props.balance / parseInt(bitcoin[0].price_usd, 10);
  }

  calculateEthereum() {
    const { ethereum } = this.props;
    if (ethereum === null || ethereum === undefined) return '';
    if (Object.keys(ethereum).length === 0) return '';
    return this.props.balance / parseInt(ethereum[0].price_usd, 10);
  }

  render() {
    return (
      <div>
        <h3 className="bitcoin-balance">Bitcoin: {this.calculateBitcoin()}</h3>
        <h3 className="ethereum-balance">
          Ethereum: {this.calculateEthereum()}
        </h3>
      </div>
    );
  }
}

export default connect(state => state, { fetchBitcoin, fetchEthereum })(Coins);
