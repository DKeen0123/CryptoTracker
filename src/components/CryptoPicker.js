import React, { Component } from 'react';

class CryptoPicker extends Component {
  constructor(props) {
    super()
  }
  populateCryptos() {
    const { cryptos } = this.props;
    if (cryptos === null || cryptos === undefined) return '';
    if (Object.keys(cryptos).length === 0) return '';
    return cryptos.map(crypto => {
      return (
        <option id={crypto.id} key={crypto.id}>
          {crypto.name}
        </option>
      );
    });
  }

  render() {
    return (
      <div>
        <h3>Choose your crypto: </h3>
        <select
          onChange={this.props.selectCrypto}
          className="drop-down"
          value={this.props.chosenCrypto}
        >
          {this.populateCryptos()}
        </select>

      </div>
    );
  }
}

export default CryptoPicker;