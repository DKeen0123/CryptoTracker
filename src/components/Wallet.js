import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Wallet extends Component {
  constructor() {
    super();

    this.state = { balance: undefined };
  }

  updateBalance = event =>
    this.setState({ balance: parseInt(event.target.value, 10) });

  render() {
    return (
      <div>
        <h3 className="balance">Balance: {this.props.balance}</h3>
        <input className="input-balance" onChange={this.updateBalance} />
      </div>
    );
  }
}

export default connect(state => {
  return { balance: state.balance };
}, null)(Wallet);
