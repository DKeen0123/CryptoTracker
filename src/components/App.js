import React from 'react';
import Wallet from './Wallet';

const App = () => {
  return (
    <div>
      <h2>CryptoTracker</h2>
      <hr />
      <Wallet />
    </div>
  );
};

export default App;
