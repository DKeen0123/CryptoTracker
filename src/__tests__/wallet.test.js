import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from '../components/wallet';

describe('Wallet', () => {
  const props = { balance: 10 };
  const wallet = shallow(<Wallet {...props} />);

  it('renders correctly', () => {
    expect(wallet).toMatchSnapshot();
  });

  it('gets the balance from props', () => {
    expect(wallet.find('.balance').text()).toEqual('Balance: 10');
  });
});
