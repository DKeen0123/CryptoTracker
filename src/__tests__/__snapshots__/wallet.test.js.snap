// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3
        className="balance"
>
        Balance: 
        10
</h3>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Balance: ",
          10,
        ],
        "className": "balance",
      },
      "ref": null,
      "rendered": Array [
        "Balance: ",
        10,
      ],
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3
          className="balance"
>
          Balance: 
          10
</h3>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Balance: ",
            10,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Balance: ",
          10,
        ],
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
